### ISSUE TITLE
TEST-04-JOURNAL: testsuite-04.bsod.sh is unstable

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#31334]
https://jenkins-systemd.apps.ocp.cloud.ci.centos.org/job/upstream-vagrant-archlinux/10698/artifact/systemd-centos-ci/artifacts_t_8yywsn/vagrant-logs.95w/vagrant-arch-testsuite.zot/TEST-04-JOURNAL_FAIL.log

**Comment by mrc0mmand** [comment]
@yuwata so, this appears to be fallout from bd1af1d5fb1, resp. from f7f062bf9f1, where sd-bsod occasionally crashes:

```
[   11.006475] testsuite-04.sh[398]: Probing EDD (edd=off to disable)... ok
[   11.009035] testsuite-04.sh[256]: + systemctl status systemd-bsod.service
[   11.011184] testsuite-04.sh[399]: × systemd-bsod.service - Displays emergency message in full screen.
[   11.014314] testsuite-04.sh[399]:      Loaded: loaded (/usr/lib/systemd/system/systemd-bsod.service; static)
[   11.017187] testsuite-04.sh[399]:     Drop-In: /run/systemd/system/systemd-bsod.service.d
[   11.019615] testsuite-04.sh[399]:              └─99-override.conf
[   11.022178] testsuite-04.sh[399]:      Active: failed (Result: core-dump) since Thu 2024-02-15 15:34:54 UTC; 5s ago
[   11.025750] testsuite-04.sh[399]:    Duration: 160ms
[   11.027865] testsuite-04.sh[399]:        Docs: man:systemd-bsod.service(8)
[   11.030191] testsuite-04.sh[399]:     Process: 351 ExecStart=/usr/lib/systemd/systemd-bsod --continuous (code=dumped, signal=ABRT)
[   11.034091] testsuite-04.sh[399]:    Main PID: 351 (code=dumped, signal=ABRT)
[   11.037285] testsuite-04.sh[399]:    Mem peak: 1.6M
[   11.039030] testsuite-04.sh[399]:         CPU: 10ms
[   11.040872] testsuite-04.sh[399]: Feb 15 15:34:54 H (emd-bsod)[351]: Found cgroup2 on /sys/fs/cgroup/, full unified hierarchy
[   11.044920] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd-bsod[351]: Assertion '!FLAGS_SET(j->flags, SD_JOURNAL_ASSUME_IMMUTABLE)' failed at src/libsystemd/sd-journal/sd-journal.c:3021, function sd_journal_wait(). Aborting.
[   11.050690] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Child 351 belongs to systemd-bsod.service.
[   11.054835] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Main process exited, code=dumped, status=6/ABRT
[   11.058765] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Failed with result 'core-dump'.
[   11.062566] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Service will not restart (restart setting)
[   11.066188] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Changed running -> failed
[   11.069341] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Unit entered failed state.
[   11.073174] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Consumed 10ms CPU time, 1.6M memory peak.
[   11.077896] testsuite-04.sh[399]: Feb 15 15:34:54 H systemd[1]: systemd-bsod.service: Releasing resources...
[   11.081429] testsuite-04.sh[249]: + echo 'Subtest /usr/lib/systemd/tests/testdata/units/testsuite-04.bsod.sh failed'
[   11.085009] testsuite-04.sh[249]: Subtest /usr/lib/systemd/tests/testdata/units/testsuite-04.bsod.sh failed
```

The journal shenanigans in the `at_exit()` handler are apparently not enough to preserve the journal, the coredump never made it outside of the test VM.

**Comment by yuwata** [comment]
Ouch. Fix is waiting in #31340.
